appveyor.yml

```diff
@@ -0,0 +1,93 @@
+version: 0.0.4-appv.{build}
+branches:
+  only:
+  - add-travis-appveyor
+  - language-server-feature
+clone_depth: 10
+init:
+- SET
+
+environment:
+  matrix:
+  # Language Server testing
+  # LTS Agent
+  - PUPPET_GEM_VERSION: "~> 4.7.0"
+    RUBY_VER: 21-x64
+    RAKE_TASK: test
+  # Latest 4.x branch
+  - PUPPET_GEM_VERSION: "~> 4.0"
+    RUBY_VER: 21-x64
+    RAKE_TASK: test
+  # Latest Puppet
+  - PUPPET_GEM_VERSION: "> 4.0.0"
+    RUBY_VER: 23-x64
+    RAKE_TASK: test
+  # Ruby style
+  - PUPPET_GEM_VERSION: "> 4.0.0"
+    RUBY_VER: 23-x64
+    RAKE_TASK: rubocop
+
+  # Extension testing
+  # Package it up to a VSIX
+  - nodejs_version: "7.4.0"
+    nodejs_arch: "x64"
+    GULP_BUILD_TASK: build
+
+matrix:
+  fast_finish: true
+  allow_failures:
+    # Ruby style
+    - PUPPET_GEM_VERSION: "> 4.0.0"
+      RUBY_VER: 23-x64
+      RAKE_TASK: rubocop
+
+install:
+- ps: |
+     Write-Host "Setting build version..."
+     $ENV:APPVEYOR_BUILD_VERSION | Out-File -FilePath 'server\lib\puppet-languageserver\VERSION' -Encoding ASCII -Confirm:$false -Force
+
+     if ($ENV:RUBY_VER -ne $null) {
+       Set-Location -Path ".\server"
+       $ENV:Path = 'C:\Ruby' + $ENV:RUBY_VER + '\bin;' + $ENV:Path
+
+       Write-Host "Ruby Version..."
+       & ruby -v
+       Write-Host "Gem Version..."
+       & gem -v
+       Write-Host "Bundle Version..."
+       & bundle -v
+       & bundle install
+
+       type Gemfile.lock
+     }
+
+     if ($ENV:nodejs_version -ne $null) {
+       Install-Product node $env:nodejs_version $env:nodejs_arch
+       Set-Location -Path ".\client"
+
+       Write-Host "Node Version..."
+       & node --version
+
+       Write-Host "Updating npm..."
+       & npm install -g npm@4 --silent
+
+       Write-Host "npm Version..."
+       & npm --version
+
+       Write-Host "Installing modules..."
+       & npm install --silent
+
+       Write-Host "Set the package.json version..."
+       & node node_modules\gulp\bin\gulp.js bump --version $ENV:APPVEYOR_BUILD_VERSION
```

Comment:
I have it changing the version so that PR builds etc. generate packages with an obviously "test" version number.  And we don't have to embed that into the package.json, CI does it for us.